pkgs/tools/package-management/poetry/unwrapped.nix

```diff
@@ -42,9 +43,36 @@
 , pythonAtLeast
 }:
 
+let
+  poetry-core-overlay = poetry-core.overridePythonAttrs (old: rec {
+    version = "1.6.1";
+    src = fetchFromGitHub {
+      owner = "python-poetry";
+      repo = "poetry-core";
+      rev = version;
+      hash = "sha256-Gc22Y2T4uO39jiOqEUFeOfnVCbknuDjmzFPZgk2eY74=";
+    };
+    nativeCheckInputs = old.nativeCheckInputs ++ [
+      tomli
+      tomli-w
+    ];
+  });
+  poetry-plugin-export-overlay = poetry-plugin-export.overridePythonAttrs(old: {
+    version = "1.4.0";
+    src = fetchFromGitHub {
+      owner = "python-poetry";
+      repo = old.pname;
+      rev = "refs/tags/${old.version}";
+      hash = "sha256-okI91Z9u5w7IHpPb9jL4Hb8+MkYJEF2qm0mqqCdyKbk=";
+    };
+    nativeBuildInputs = ((builtins.filter (i: i.pname != poetry-core-overlay.pname) old.nativeBuildInputs) ++ [
+      poetry-core-overlay
+    ]);
+  });
```

Comment:
I believe these overrides would be better to add to the python3 packageOverrides in the `default.nix` file:

https://github.com/NixOS/nixpkgs/pull/241458/files#diff-1e6bf80c8cd719f30a6f001e8aa296035de88b1cdb8840b953cd66b5f203a44eR8

I based this [commit](https://github.com/considerate/nixpkgs/commit/809696c8a33827dcba78d305c7449f85dc8242da) on this PR and I think it simplifies the build a bit compared to this PR.

Reply:
I've incorporated your suggestions into the PR. I was unaware that the override would behave cyclically even back into the `poetry-plugin-export`'s `nativeBuildInputs`. This is far cleaner than the prior version- thank you ^^.